ui.frontend/src/view/FormFieldBase.js

```diff
@@ -118,6 +120,40 @@ export default class FormFieldBase extends FormField {
         }
     }
 
+    registerWidgetEventListeners() {
+        const guideBridge = window.guideBridge;
+        const onfocus = (e) => {
+            const formContainerPath = this.formContainer.getPath();
+            const formName = this.#getFormName();
+            const panelName = this.#getPanelName();
+            const fieldName = this._model._jsonModel.name;
+            const eventPayload = {
+                formName,
+                fieldName,
+                panelName
+            };
+            guideBridge.trigger(this.GUIDE_ELEMENT_FOCUS_CHANGED, eventPayload, formContainerPath);
+        }
+        const widget = this.getWidget();
+        if (widget) {
+            if (widget.length && widget.length > 1) {
+                for (let opt of widget) {
+                    opt.onfocus = onfocus;
+                }
+            } else {
+                widget.onfocus = onfocus;
+            }
+        }
+    }
+
+    #getFormName() {
+        return this.formContainer.getPath().replace('/jcr:content/guideContainer', '')
```

Comment:
we should get this info from the model. this is hacky.

Reply:
`this.parentView.getModel().name`

Reply:
We are not getting any information regarding the form name except the encoded ID

Reply:
we should get the form name also.. this will not work.